botnim/vector_store/vector_store_es.py

```diff
@@ -0,0 +1,144 @@
+from .vector_store_base import VectorStoreBase
+from elasticsearch import Elasticsearch
+from openai import OpenAI
+import os
+from botnim.config import get_logger
+
+logger = get_logger(__name__)
+
+class VectorStoreES(VectorStoreBase):
+    """
+    Vector store for Elasticsearch
+    """	
+    def __init__(self, config, config_dir, production, 
+                 es_host, es_username, es_password, 
+                 es_timeout=30, verify_certs=False):
+        super().__init__(config, config_dir, production)
+        
+        # Initialize Elasticsearch client
+        es_kwargs = {
+            'hosts': [es_host],
+            'basic_auth': (es_username, es_password or os.getenv('ELASTIC_PASSWORD')),
+            'request_timeout': es_timeout,
+            'verify_certs': verify_certs,
+            'ssl_show_warn': False
+        }
+        
+        self.es_client = Elasticsearch(**es_kwargs)
+        self.openai_client = OpenAI(api_key=os.getenv('OPENAI_API_KEY'))
+        self.init = False
+        
+        # Verify connection
+        try:
+            if not self.es_client.ping():
+                raise ConnectionError("Could not ping Elasticsearch")
+            info = self.es_client.info()
+            logger.info(f"Connected to Elasticsearch version {info['version']['number']}")
+        except Exception as e:
+            logger.error(f"Failed to connect to Elasticsearch: {str(e)}")
+            raise ConnectionError(f"Could not connect to Elasticsearch: {str(e)}")
+
+    def get_or_create_vector_store(self, context, context_name, replace_context):
+        ret = None # return value 
+        vs_name = self.env_name(self.config['name']).lower().replace(' ', '_')
```

Comment:
we need to include the context name as well here, as we would want each context to be in a separate index.